tests/framework/extensions/provisioning/creates.go

```diff
@@ -227,22 +227,30 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 		}
 	}
 
-	if clustersConfig.Hardened && clustersConfig.KubernetesVersion <= string(provisioninginput.HardenedKubeVersion) {
-		err = hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
-		}
+	if clustersConfig.Hardened {
+		if strings.Contains(clustersConfig.KubernetesVersion, "k3s") {
+			err = k3sHardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode, clustersConfig.KubernetesVersion)
+			if err != nil {
+				return nil, err
+			}
 
-		hardenCluster := clusters.HardenK3SRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			hardenCluster := clusters.HardenK3SClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
+			if err != nil {
+				return nil, err
+			}
 
-		_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
-		if err != nil {
-			return nil, err
-		}
+		} else {
+			err = rke2Hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
+			if err != nil {
+				return nil, err
+			}
 
-		err = hardening.PostHardeningConfig(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
+			hardenCluster := clusters.HardenRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
nit; this is shared between the 'if' and 'else' , so could be pulled out and placed outside of the conditional

Reply:
I agree, `else if` might be safer

Reply:
Addressed

Reply:
Yeah that's what I see can you elaborate Caleb?

Reply:
just the last 4 lines are the same in each if/else, so I thought it could be pulled out since they are shared between both cases.

Reply:
Ah I see your point, so hardened the nodes depending on cluster type, and then update it at the end. Can we do that Markus instead of repeating the same code over?

Reply:
Got it, made the change and tested locally that all is well.